src/connection/connection.go

```diff
@@ -0,0 +1,180 @@
+func (c SaramaClient) Coordinator(groupID string) (Broker, error) {
+	return c.Inner.Coordinator(groupID)
+}
+
+// Leader wraps the sarama.Client.Leader() function
+func (c SaramaClient) Leader(topic string, partition int32) (Broker, error) {
+	return c.Inner.Leader(topic, partition)
+}
+
+// Close wraps the sarama.Client.Close() function
+func (c SaramaClient) Close() error {
+	return c.Inner.Close()
+}
+
+// GetOffset wraps the sarama.Client.GetOffset() function
+func (c SaramaClient) GetOffset(topic string, partition int32, offset int64) (int64, error) {
+	return c.Inner.GetOffset(topic, partition, offset)
+}
+
+// Broker is an interface for mocking
+type Broker interface {
+	Connected() (bool, error)
+	FetchOffset(*sarama.OffsetFetchRequest) (*sarama.OffsetFetchResponse, error)
+	Fetch(*sarama.FetchRequest) (*sarama.FetchResponse, error)
+	Open(*sarama.Config) error
+	DescribeGroups(*sarama.DescribeGroupsRequest) (*sarama.DescribeGroupsResponse, error)
+	ListGroups(*sarama.ListGroupsRequest) (*sarama.ListGroupsResponse, error)
+	Close() error
+}
+
+// MockClient is a mock implementation of the Client interface
+type MockClient struct {
+	mock.Mock
+}
+
+// Brokers is a mocked implementation of the sarama.Client.Brokers() method
+func (m MockClient) Brokers() []Broker {
+	args := m.Called()
+	return args.Get(0).([]Broker)
+}
+
+// Topics is a mocked implementation of the sarama.Client.Topics() method
+func (m MockClient) Topics() ([]string, error) {
+	args := m.Called()
+	return args.Get(0).([]string), args.Error(1)
+}
+
+// Partitions is a mocked implementation of the sarama.Client.Partitions() method
+func (m MockClient) Partitions(topic string) ([]int32, error) {
+	args := m.Called(topic)
+	return args.Get(0).([]int32), args.Error(1)
+}
+
+// RefreshCoordinator is a mocked implementation of the sarama.Client.RefreshCoordinator() method
+func (m MockClient) RefreshCoordinator(groupID string) error {
+	args := m.Called(groupID)
+	return args.Error(0)
+}
+
+// Coordinator is a mocked implementation of the sarama.Client.Coordinator() method
+func (m MockClient) Coordinator(groupID string) (Broker, error) {
+	args := m.Called(groupID)
+	return args.Get(0).(Broker), args.Error(1)
+}
+
+// Leader is a mocked implementation of the sarama.Client.Leader() method
+func (m MockClient) Leader(topic string, partition int32) (Broker, error) {
+	args := m.Called(topic, partition)
+	return args.Get(0).(Broker), args.Error(1)
+}
+
+// Close is a mocked implementation of the sarama.Client.Close() method
+func (m MockClient) Close() error {
+	args := m.Called()
+	return args.Error(0)
+}
+
+// GetOffset is a mocked implementation of the sarama.Client.GetOffset() method
+func (m MockClient) GetOffset(topic string, partition int32, offset int64) (int64, error) {
+	args := m.Called(topic, partition, offset)
+	return args.Get(0).(int64), args.Error(1)
+}
+
+// MockBroker is a mock implementation of the Broker interface
+type MockBroker struct {
```

Comment:
`MockBroker` should be in the same file as `MockClient`. Best to keep test code separate.